Code-4-Community/speak-for-the-trees-frontend
# Sk.cl.upload site image

## Checklist


## Why

Resolves #241 

This ticket allows the end users to upload their tree image to their adopted tree.  Previously, there was no way of uploading tree images through the site and thus this will come in handy to users who want to upload their tree photos to share with other users. 

## This PR
This change adds an upload image component to the tree pages. Users are able to upload tree images that are converted to b64. The upload image component with its API client 'uploadImage' posts the uploaded image and siteEntryId to /api/v1/protected/sites/site_image/${siteEntryId}. Note that you have to adopt the tree to be able to upload a tree image. 

## Screenshots

<img width="657" alt="Screen Shot 2023-10-22 at 3 48 12 PM" src="https://github.com/Code-4-Community/speak-for-the-trees-frontend/assets/102868495/39ac2462-7adf-446b-a1fb-b3820f8c2845">
<img width="783" alt="Screen Shot 2023-10-22 at 3 48 53 PM" src="https://github.com/Code-4-Community/speak-for-the-trees-frontend/assets/102868495/1350abf2-9019-4a3c-98f7-d74493065d04">
<img width="667" alt="Screen Shot 2023-10-22 at 3 49 13 PM" src="https://github.com/Code-4-Community/speak-for-the-trees-frontend/assets/102868495/228a55d4-d59d-45b4-bc2d-051176416d79">

## Screen recording

https://github.com/Code-4-Community/speak-for-the-trees-frontend/assets/30560773/d784513d-d8bb-4986-b3f0-edea9fcc0527


## Verification Steps
To check if this upload component works, try uploading an image and go to AWS to see if the image shows up in folder 'sfft-user-upload' -  (the content you defined in 'aws_s3_upload_dir' in server.properties from the backend). Additionally, there are three tests we could run in protectedApiClient.test to test upload image API client.
